### PR TITLE
Tutorial structure: remove deduplication, includify

### PR DESCRIPTION
**Pull request by brenthuisman**
* added notes re literal_include to python examples
* merged changes in #1504 
* deduped and includified (https://github.com/arbor-sim/arbor/pull/1558) tutorials.
* First divergence of tutorial and code spotted!
* Added some contrib documentation for examples and docs.

### CODE REVIEW

File: doc/tutorial/single_cell_detailed.rst [hunk 1]
```diff
@@ -18,8 +18,13 @@ complex single cell model, and go through the process in more detail.
...
    7. Running a simulation and visualising the results using a :class:`arbor.single_cell_model`.
 
+.. _tutorialsinglecellswc-cell:
+
+The cell
+********
+
 We start by building the cell. This will be a :ref:`cable cell <cablecell>` with complex
-geometry and dynamics which can be constructed from 3 components:
+geometry and dynamics which is be constructed from 3 components:
```
Comment (noraabiakar) [inline doc/tutorial/single_cell_detailed.rst:27]: remove 'be'

File: doc/tutorial/single_cell_detailed.rst [hunk 2]
```diff
@@ -28,31 +33,12 @@ geometry and dynamics which can be constructed from 3 components:
    The decor also includes hints about how the cell is to be modelled under the hood, by
    splitting it into discrete control volumes (CV).
 
-Next, we construct a :class:`arbor.single_cell_model`. This model takes care of a lot of details
```
Comment (noraabiakar) [inline doc/tutorial/single_cell_detailed.rst:31]: Why was this removed?

File: doc/tutorial/single_cell_recipe.rst [hunk 3]
```diff
@@ -12,86 +12,34 @@ and :class:`arbor.simulation` instead of a :class:`arbor.single_cell_model`.
...
+   :lines: 4,8-23
 
 The recipe
 ----------
 
-The :class:`arbor.single_cell_model` of the previous example created a :class:`arbor.recipe` under
-the hood, and abstracted away the details so we were unaware of its existence.
+The :class:`arbor.single_cell_model` of the previous example creates a :class:`arbor.recipe` under
+the hood, and abstracts away a few details that you may want control over in more complex simulations.
+Let's go into which we need to go into so we were unaware of its existence.
```
Comment (noraabiakar) [inline doc/tutorial/single_cell_recipe.rst:33]: I don't understand this sentence.

File: python/example/network_ring_mpi.py [hunk 4]
```diff
@@ -108,7 +109,7 @@ def global_properties(self, kind):
         return self.props
 
 # (11) Instantiate recipe
-ncells = 50
+ncells = 500
```
Comment (noraabiakar) [inline python/example/network_ring_mpi.py:112]: Why increase the number of cells?

File: doc/tutorial/network_ring.rst [hunk 5]
```diff
@@ -16,35 +16,18 @@ In this example, a small *network* of cells, arranged in a ring, will be created
 The cell
 ********
 
-Step **(1)** shows how a simple cell with a dendrite is created. We construct the following :term:`morphology` and label the soma and dendrite:
+Step **(1)** shows how a function that returns a simple cell with a dendrite is created.
```
Comment (noraabiakar) [inline doc/tutorial/network_ring.rst:19]: This is quite confusing. Maybe:` Step **(1)** shows a function that creates a simple cell with a dendrite.`

**Review by noraabiakar — CHANGES_REQUESTED** [review]
Looks good, I just have some minor comments.

File: doc/tutorial/single_cell_detailed.rst (hunk 2 above)
Comment (brenthuisman) [reply doc/tutorial/single_cell_detailed.rst:31]: Because the introduction of the `single_cell_model` is in `single_cell_model.rst`, and we wanted some of the duplication that exists between the tutorials.

File: doc/tutorial/single_cell_recipe.rst (hunk 3 above)
Comment (brenthuisman) [reply doc/tutorial/single_cell_recipe.rst:33]: Me neither :P I hope the rewrite is better ;)

File: python/example/network_ring_mpi.py (hunk 4 above)
Comment (brenthuisman) [reply python/example/network_ring_mpi.py:112]: The tutorial text actually mentions 500 cells, but the final plot (and the code in the `.py`) used `ncells=50` to not generate an overcrowded plot. See also the caption of the plot.

File: doc/tutorial/network_ring.rst (hunk 5 above)
Comment (brenthuisman) [reply doc/tutorial/network_ring.rst:19]: :+1:

File: doc/tutorial/single_cell_model.rst [hunk 6]
```diff
@@ -106,16 +88,18 @@ Arbor provides an interface for constructing single cell models with the
...
-    # (5) Make single cell model.
-    m = arbor.single_cell_model(cell)
+1. It holds the **global properties** of the model
+2. It registers **probes** on specific locations on the cell to measure the voltage.
+3. It **runs** the simulation.
+4. It collects **spikes** from spike detectors and voltage **traces** from registered probes.
 
-    # (6) Attach voltage probe sampling at 10 kHz (every 0.1 ms).
-    m.probe('voltage', '"midpoint"', frequency=10)
+Right now, we'll only be setting a probe and run the simulation.
```
Comment (noraabiakar) [inline doc/tutorial/single_cell_model.rst:98]: `running`

**Review by noraabiakar — COMMENTED** [review]
I just missed one typo

**Review by noraabiakar — APPROVED** [review]
